app/controllers/lead_providers/school_details_controller.rb

```diff
@@ -3,8 +3,22 @@
 module LeadProviders
   class SchoolDetailsController < ::LeadProviders::BaseController
     def show
-      @school = School.eligible.find(params[:id])
-      @selected_cohort = Cohort.find(params[:selected_cohort_id])
+      @school = find_school
+      @selected_cohort = selected_cohort_or_current
+    end
+
+  private
+
+    def selected_cohort_or_current
+      Cohort.find_by(id: params[:selected_cohort_id]) || Cohort.current
+    end
+
+    def find_school
+      year = selected_cohort_or_current.start_year
+      search_scope = School.eligible
+
+      search_scope = search_scope.partnered_with_lead_provider(current_user.lead_provider.id, year) if current_user.lead_provider?
```

Comment:
Is the `curret_user` ever not going to be a lead provider? Given that we inherit from `::LeadProviders::BaseController`?

Reply:
All the tests pass, I'll remove the admin check in the base controller then and simplify that find school method